packages/server/database/types/TemplateDimension.ts

```diff
@@ -0,0 +1,36 @@
+import shortid from 'shortid'
+
+export interface TemplateDimensionInput {
+  id?: string
+  name: string
+  teamId: string
+  templateId?: string
+  scaleId?: string
+  createdAt?: Date
+  updatedAt?: Date
+  deletedAt?: Date
+}
+
+export default class TemplateDimension {
+  id: string
+  name: string
+  teamId: string
+  templateId: string | null
+  scaleId: string | null
```

Comment:
non null?

Reply:
yeah i wasn't 100% sure about these non-nulls. my thinking was the entity should still be able to persist without those fk ids, even if the initial endpoints require those properties. what do you think? we can always make them non null to begin with, then adjust if requirements change.

Reply:
let's do it! i'm always in favor of having the strictest possible types, helps errors pop up so much sooner.
on the big picture, a dimension can't function without a scale or a template, so I'd say it's gotta have those. We could have a `DimensionInProgress` class that had them as nullable, but i don't think the UI requires that